### PR TITLE
Fix Watchdog port being used as string

### LINKED ISSUE #2902: New error after some changes to the configuration and updating the Mosquitto Add-on
### The problem

I recently made some changes to the configuration of HA and the Add-ons (I started to use Secrets and split up the configuration.yaml file, the Config Check was OK), and I noticed the error below in the Supervisor logs afterwards 
(it doesn't show up in the settings page). 
An identical error appears every few minutes.
Everything seems to be working fine except of the error.

### What is version of Home Assistant Core has the issue?

2021.5.4

### What was the last working version of Home Assistant Core?

2021.5.4

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

_No response_

### Link to integration documentation on our website

_No response_

### Example YAML snippet

```yaml
##########################
##  CONFIGURATION.YAML  ##
##########################


# Configure a default setup of Home Assistant (frontend, api, etc)
default_config:


logger:
  default: info


# Text to speech
tts:
  - platform: google_translate



# Includes
group: !include groups.yaml
automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml


sensor: !include sensor.yaml
light: !include light.yaml


homeassistant:
  customize: !include customize.yaml
  


wake_on_lan:



# TV Samsung
samsungtv:
  - host: 192.168.1.7
    name: TV Samsung
    turn_on_action:
      - service: wake_on_lan.send_magic_packet
        data:
          mac: "*redacted*"



# HomeKit
homekit: !include homekit_config.yaml



###############################
# Mosquitto Add-on configuration
###############################

logins:
  - username: '!secret Mosquitto_User'
    password: '!secret Mosquitto_Passw'
customize:
  active: false
  folder: mosquitto
certfile: fullchain.pem
keyfile: privkey.pem
require_certificate: false



###############################
# DuckDNS Add-on configuration
###############################


lets_encrypt:
  accept_terms: true
  certfile: fullchain.pem
  keyfile: privkey.pem
token: '!secret DuckDNS_Token'
domains:
  - *redacted*.duckdns.org
aliases: []
seconds: 600



###############################
# Samba Add-on configuration
###############################

workgroup: WORKGROUP
username: '!secret Samba_User'
password: '!secret Samba_Passw'
interface: ''
allow_hosts:
  - 10.0.0.0/8
  - 172.16.0.0/12
  - 192.168.0.0/16
veto_files:
  - ._*
  - .DS_Store
  - Thumbs.db
  - icon?
  - .Trashes
compatibility_mode: false

#########################
```

### Anything in the logs that might be useful for us?

```txt
21-05-19 09:05:54 ERROR (MainThread) [asyncio] Task exception was never retrieved
future: <Task finished name='Task-878' coro=<Scheduler._run_task.<locals>._wrap_task() done, defined at /usr/src/supervisor/supervisor/misc/scheduler.py:58> exception=TypeError('an integer is required (got type str)')>
Traceback (most recent call last):
  File "/usr/src/supervisor/supervisor/misc/scheduler.py", line 62, in _wrap_task
    await task.coro_callback()
  File "/usr/src/supervisor/supervisor/misc/tasks.py", line 416, in _watchdog_addon_application
    if addon.in_progress or await addon.watchdog_application():
  File "/usr/src/supervisor/supervisor/addons/addon.py", line 479, in watchdog_application
    return await self.sys_run_in_executor(check_port, self.ip_address, port)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/supervisor/supervisor/utils/__init__.py", line 43, in check_port
    result = sock.connect_ex((str(address), port))
TypeError: an integer is required (got type str)
```

### Additional information

image

image

### PR DESCRIPTION
**Pull request by frenck**
## Proposed change

When using a Watchdog URI formatted as `tcp://`, the port is a string, which:

```py
result = sock.connect_ex((str(address), port))
```

Doesn't like. This PR ensures the regex port match result is cast to an integer.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #2902
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/addons/addon.py [hunk 1]
```diff
@@ -453,7 +453,7 @@ async def watchdog_application(self) -> bool:
         application = RE_WATCHDOG.match(url)
 
         # extract arguments
-        t_port = application.group("t_port")
+        t_port = int(application.group("t_port"))
```
Comment (ludeeus) [inline supervisor/addons/addon.py:456]: Shouldn't this be on L469 instead?

Comment (frenck) [reply supervisor/addons/addon.py:456]: Why? we extract the port number here and we should make sure it is a number at any point. This is as close to the source as possible

**Review by pvizeli — APPROVED** [review]

**Comment by dnedeltchev** [comment]
Hey guys, so if I understand correctly, supervisor 2021.05.0 will fix this? So we have to wait for it to be released? Although it looks like it has been released but no notification on my HA that there is an update.

**Comment by frenck** [comment]
@dnedeltchev This isn't in any release yet. And yes, you'll have to wait until it is released and promoted to stable.

**Pull request merged**